src/client/datascience/jupyter/kernels/kernel.ts

```diff
@@ -150,6 +164,27 @@ export class Kernel implements IKernel {
             }
         }
     }
+    private async trackNotebookCellPerceivedColdTime(
+        stopWatch: StopWatch,
+        notebookPromise: Promise<INotebook | undefined>,
+        executionPromise: Promise<unknown>
+    ): Promise<void> {
+        if (this.perceivedJupyterStartupTelemetryCaptured) {
+            return;
+        }
+        const notebook = await notebookPromise;
+        if (!notebook) {
+            return;
+        }
+        // Setup telemetry
+        if (!this.perceivedJupyterStartupTelemetryCaptured) {
+            this.perceivedJupyterStartupTelemetryCaptured = true;
+            sendTelemetryEvent(Telemetry.PerceivedJupyterStartupNotebook, stopWatch.elapsedTime);
+            executionPromise.finally(() =>
+                sendTelemetryEvent(Telemetry.StartExecuteNotebookCellPerceivedCold, stopWatch.elapsedTime)
```

Comment:
Here's an example of where I'm not sure if this should be a 'kernel' event or just a normal telemetry event.

Reply:
This is existing telemetry, hence didn't want to mess around with that one.
I created a new telemetry event that will get used in all remote, local, jupyter, raw, notebooks & interactive:

* Starting notebook, interrupting, restarting & switching kernels (capture success & failures) along with all of the information we are interested in (error type, etc)


That should give us what we need to identify issues.
In a serparate PR i will look into adding this to all other events as well.